share/functions/prompt_pwd.fish

```diff
@@ -24,11 +24,24 @@ function prompt_pwd --description 'short CWD for the prompt'
     set -q fish_prompt_pwd_full_dirs
     or set -l fish_prompt_pwd_full_dirs 1
 
+    set -ql _flag_p
+    and set -l fish_prompt_pwd_dir_depth $_flag_p
+
+    set -q fish_prompt_pwd_dir_depth
+    or set -l fish_prompt_pwd_dir_depth 0
+
     for path in $argv
         # Replace $HOME with "~"
         set -l realhome (string escape --style=regex -- ~)
         set -l tmp (string replace -r '^'"$realhome"'($|/)' '~$1' $path)
 
+        # Shorten to at most $fish_prompt_pwd_dir_depth components, unless set to 0.
+        if test "$fish_prompt_pwd_dir_depth" -gt 0
```

Comment:
Right, this option makes sense.
I wonder if we should prefix shortened paths with an ellipsis,
that is

	set tmp "$(string shorten --left -m 1 ..)/$tmp"

	set fish_prompt_pwd_dir_depth 1
	# …/fish-shell
	set fish_prompt_pwd_dir_depth 2
	# …/git/fish-shell

though I guess that would stray from what bash does

Reply:
The user can do that pretty easily by querying the value of `prompt_pwd` in their prompt: prepending `...` if `(string split / (prompt_pwd))[1]` is ` ` or `~`. The other way around seems like a bit more of a pain.